.github/workflows/commit.yaml

```diff
@@ -62,13 +62,8 @@ jobs:
       - name: "Run e2e tests using the `getenvoy` binary"
         run: make e2e
 
-      - name: "Generate test coverage report"
+      - name: "Generate and upload coverage report"
         if: runner.os == 'Linux'  # no need to do this per operating system
-        run: make coverage
-
-      - name: "Upload test coverage report"
-        uses: actions/upload-artifact@v2
-        if: runner.os == 'Linux'  # no need to do this per operating system
-        with:
-          name: coverage
-          path: build/coverage
+        run: |  # This uses the $CODECOV_TOKEN repository secret
+          make coverage
+          bash <(curl -s https://codecov.io/bash) -f coverage/coverage.txt
```

Comment:
this is the canonical way to upload, except we are specifying a separate directory instead of letting coverage fall into the root https://github.com/codecov/example-go/blob/master/.github/workflows/ci.yml#L18